lsp/nls/src/requests/formatting.rs

```diff
@@ -0,0 +1,69 @@
+use std::process;
+
+use lsp_server::{ErrorCode, RequestId, Response, ResponseError};
+use lsp_types::{DocumentFormattingParams, Position, Range, TextEdit};
+
+use crate::server::{self, Server};
+
+pub fn handle_format_document(
```

Comment:
I think it could be helpful to document public API functions, as a rule of thumb.